Adding information for handling S3 Sink multipart upload aborted parts (#385)
Aiven-Open/cloud-storage-connectors-for-apache-kafka: s3-sink-connector/README.md

- To use SSE-KMS set to `aws:kms`
- To use DSSE-KMS set to `aws:kms:dsse`

> ### Incomplete Multipart uploads

Reviewer: ```suggestion
### Cleaning temporary files from failed multipart uploads
```
I don't know if this is a better title, but something a bit more active would also be more clear!

The S3 Sink Connector uploads the files using the S3 Mutli part upload API for improved performance and handling large file sizes.
Occasionally the API can throw an exception or the connector can fail to complete a multipart upload.

> This can leave "Parts" of the multipart upload on S3 waiting to complete taking up unnecessary space.

Reviewer: ```suggestion
This can leave "parts" of a failed multipart upload taking up unnecessary space.
```

To handle these incomplete parts AWS recommends setting up a Lifecycle rule to delete old parts that weren't completed as described in this excellent (blog post)[https://aws.amazon.com/blogs/aws-cloud-financial-management/discovering-and-deleting-incomplete-multipart-uploads-to-lower-amazon-s3-costs/].

> Or if you would prefer to work through the official documentation it is available (here)[https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpu-abort-incomplete-mpu-lifecycle-config.html]

Reviewer: ```suggestion
Alternatively, if you would prefer to work through the official documentation it is available (here)[https://docs.aws.amazon.com/AmazonS3/latest/userguide/mpu-abort-incomplete-mpu-lifecycle-config.html]
```
